Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1019,8 +1019,10 @@ void RenderElement::styleDidChange(StyleDifference diff, const RenderStyle* oldS
     auto registerImages = [this](auto* style, auto* oldStyle) {
         if (!style && !oldStyle)
             return;
-        updateFillImages(oldStyle ? &oldStyle->protectedBackgroundLayers().get() : nullptr, style ? &style->protectedBackgroundLayers().get() : nullptr);
-        updateFillImages(oldStyle ? &oldStyle->protectedMaskLayers().get() : nullptr, style ? &style->protectedMaskLayers().get() : nullptr);
+        if ((style && style->backgroundLayers().hasImage()) || (oldStyle && oldStyle->backgroundLayers().hasImage()))
```

Comment:
`hasImage()` has to walk the layer list, so is this really faster?

Reply:
The vast majority of cases have a single FillLayer with no image, so yes.

I guess in rare cases where you have many elements, each with multiple background layers it could be slower.